apps/games/g_snake/main.py

```diff
@@ -0,0 +1,110 @@
+from ui import Canvas
+from time import sleep
+from ui import DialogBox, ffs
+from helpers import ExitHelper
+from threading import Event, Lock
+from random import randint
+
+menu_name = "Snake"
+
+i = None #Input device
+o = None #Output device
+do_exit = None
+
+snake = [(10, 10), (11, 10), (12, 10)]
+lose = False
+direction = "right"
+apple = False
+
+
+def init_app(input, output):
+	#Gets called when app is loaded
+	global i, o, c
+	i = input; o = output; c = Canvas(o)
+
+	
+def callback():
+	#Gets called when app is selected from menu
+	start_game()
+	#pass
+
+def avancer():
+	snake.remove(snake[0])
+	if direction == "right":
+		snake.append((snake[-1][0]+1, snake[-1][1]))
+	elif direction == "left":
+		snake.append((snake[-1][0]-1, snake[-1][1]))
+	elif direction == "down":
+		snake.append((snake[-1][0], snake[-1][1]+1))
+	else:
+		snake.append((snake[-1][0], snake[-1][1]-1))
+
+
+def set_keymap():
+	keymap = {"KEY_LEFT": lambda:make_a_move("left"),
+			  "KEY_RIGHT": lambda:make_a_move("right"),
+			  "KEY_UP": lambda:make_a_move("up"),
+			  "KEY_DOWN": lambda:make_a_move("down"),
+			  "KEY_ENTER": confirm_exit}
+	i.stop_listen()
+	i.set_keymap(keymap)
+	i.listen()
+
+def confirm_exit():
+	choices = ["y", "n"]
+	choice = DialogBox(choices, i, o, message="Exit the game?").activate()
+	if choice is True:
+		do_exit.set()
+	else:
+		start_game()
```

Comment:
`confirm_exit()` is called from another (input callback processing) thread - so, calling `start_game()` from that thread is not what you actually want, since `start_game` is already running from another thread and doesn't stop when you press Enter and `confirm_exit` is executed (that's why I used `Lock` in 2048).